packages/grid_client/src/high_level/twinDeploymentHandler.ts

```diff
@@ -412,6 +412,42 @@ class TwinDeploymentHandler {
     return await this.deploymentFactory.fromObj(deployment);
   }
 
+  private async handleGatewayWorkloads(
+    workloads: Workload[],
+    operation: Operations,
+  ): Promise<{
+    nameExtrinsics: ExtrinsicResult<Contract>[];
+    deletedExtrinsics: ExtrinsicResult<number>[];
+  }> {
+    const nameExtrinsics: ExtrinsicResult<Contract>[] = [];
+    const deletedExtrinsics: ExtrinsicResult<number>[] = [];
+
+    for (const workload of workloads) {
+      if (workload.type === WorkloadTypes.gatewaynameproxy) {
+        events.emit("logs", `Check the name contract for the workload with name: ${workload.name}`);
+        if (operation === Operations.delete) {
+          const extrinsic = await this.deleteNameContract(workload.data["name"]);
+          if (extrinsic) deletedExtrinsics.push(extrinsic);
+        } else {
+          const extrinsic = await this.createNameContract(workload.data["name"]);
+          nameExtrinsics.push(extrinsic);
+        }
+      } else if (workload.type === WorkloadTypes.gatewayfqdnproxy) {
+        events.emit("logs", `Check the name contract for the FQDN workload with name: ${workload.name}`);
+        const contractName = workload.data["fqdn"].replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
```

Comment:
we should store the full fqdn name and do not remove anything

Reply:
I think we shouldn't as tf chain doesn't support the characters like dots for example